### PR TITLE
Add a recipe for immer

### PR DESCRIPTION
**Pull request by rilerez**
Specify library name and version:  **immer/0.6.2**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/immer/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+sources:
+  "0.6.2":
+    url: "https://github.com/arximboldi/immer/archive/refs/tags/v0.6.2.tar.gz"
```
Comment (madebr) [inline recipes/immer/all/conandata.yml:3]:
```suggestion
    url: "https://github.com/arximboldi/immer/archive/refs/tags/v0.6.2.tar.gz"
    sha256: "c3bb8847034437dee64adacb04e1e0163ae640b596c582eb4c0aa1d7c6447cd7"
```

File: recipes/refl-cpp/all/conandata.yml [hunk 2]
```diff
@@ -1,4 +0,0 @@
-sources:
-  "0.12.1":
```
Comment (madebr) [inline recipes/refl-cpp/all/conandata.yml:2]: You accidentally removed some files in `refl-cpp`.
Please undo this :smile:

Comment (rilerez) [reply recipes/refl-cpp/all/conandata.yml:2]: noooooooo

**Review by LHLaurini — CHANGES_REQUESTED** [review]
It seems you have removed the recipe for `refl-cpp` and created one for `immer` in the wrong directory (/ instead of /recipes).

File: recipes/immer/all/conandata.yml (hunk 1 above)
Comment (LHLaurini) [reply recipes/immer/all/conandata.yml:3]: @rilerez, did you forget to use this patch?

File: recipes/immer/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    def package_id(self):
+        self.info.header_only()
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "8",
+            "clang": "8",
+        }
```
Comment (LHLaurini) [inline recipes/immer/all/conanfile.py:43]:
```suggestion
        # Reference: https://en.cppreference.com/w/cpp/compiler_support/14
        return {
            "apple-clang": "5.1",
            "clang": "3.4",
            "gcc": "5",
            "intel": "17",
            "sun-cc": "5.15",
            "Visual Studio": "15"
        }
```

I suggest using these versions instead. Taken from [another recipe](https://github.com/conan-io/conan-center-index/blob/master/recipes/rmlui/3.3/conanfile.py).

**Review by LHLaurini — CHANGES_REQUESTED** [review]
Just a few details. Other than that, looks good. Also, avoid force-pushing unless necessary, it can erase the history.

**Comment by LHLaurini** [comment]
Guess that was too old. Try increasing the GCC version a little.

**Comment by LHLaurini** [comment]
Please, also accept [this](https://github.com/conan-io/conan-center-index/pull/5522#discussion_r632109515).

**Review by madebr — COMMENTED** [review]
Please add https://github.com/conan-io/conan-center-index/pull/5522/files#r632109515 and I think we're good

**Review by LHLaurini — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**